miner/worker.go

```diff
@@ -696,6 +696,10 @@ func (w *worker) updateSnapshot() {
 }
 
 func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Address) ([]*types.Log, error) {
+	// Make sure there's only one tx per block
+	if w.current != nil && len(w.current.txs) > 0 {
+		return nil, core.ErrGasLimitReached
```

Comment:
Is it easy to add another error type? My guess is no, but worth a try maybe!

Reply:
We could, but it'd function the same as this one. We're basically saying the block is full, and that's what this one signifies. 🤷